core/controllers/classifier.py

```diff
@@ -123,3 +123,30 @@ def post(self):
         classifier_services.mark_training_job_complete(job_id)
 
         return self.render_json({})
+
+
+class NextJobHandler(base.BaseHandler):
+    """This handler fetches next job in job queue and sends back job_id,
+    algorithm_id and training data to the VM.
+    """
+    REQUIRE_PAYLOAD_CSRF_CHECK = False
+
+    def post(self):
+        """Handles POST requests. """
+        signature = self.payload.get('signature')
+        vm_id = self.payload.get('vm_id')
+
+        if vm_id == feconf.DEFAULT_VM_ID and not feconf.DEV_MODE:
+            raise self.UnauthorizedUserException
+        if not verify_signature(vm_id, vm_id, signature):
```

Comment:
I thought we agreed on having empty message. So extract `message` from payload and use message to verify the signature. Why are you still using `vm_id`?